WordPress/Classes/ViewRelated/Blog/My Site/NoSitesView.swift

```diff
@@ -0,0 +1,188 @@
+            }
+        }
+    }
+}
+
+// MARK: - Add new site
+
+extension NoSitesView {
+
+    /// If the account can't add a self-hosted site, launch site creation.
+    ///
+    /// Otherwise, show options to create a WordPress.com site or add a self-hosted site.
+    /// In compact size classes, the system shows a dialog sheet with a cancel button.
+    /// In regular size classes, the system shows a popover without a cancel button.
+    ///
+    func handleAddNewSiteButtonTapped() {
+        WPAnalytics.track(.mySiteNoSitesViewActionTapped)
+
+        guard addNewSiteConfiguration.canCreateWPComSite else {
+            return
+        }
+
+        guard addNewSiteConfiguration.canAddSelfHostedSite else {
+            addNewSiteConfiguration.launchSiteCreation()
+            return
+        }
+
+        self.isShowingDialog = true
+    }
+
+    @ViewBuilder private func makeAddNewSitesDialog() -> some View {
+        Button(Strings.createWPComSite) {
+            addNewSiteConfiguration.launchSiteCreation()
+        }
+        Button(Strings.addSelfHostedSite) {
+            addNewSiteConfiguration.launchLoginForSelfHostedSite()
+        }
+    }
+}
+
+extension NoSitesView {
+    private enum Strings {
+        static let title = NSLocalizedString("mySite.noSites.title", value: "You don't have any sites", comment: "Message title for when a user has no sites.")
+        static let description = NSLocalizedString("mySite.noSites.description", value: "Create a new site for your business, magazine, or personal blog; or connect an existing WordPress installation.", comment: "Message description for when a user has no sites.")
+        static let addNewSite = NSLocalizedString("mySite.noSites.button.addNewSite", value: "Add new site", comment: "Button title. Displays a screen to add a new site when tapped.")
+        static let accountAndSettings = NSLocalizedString("mySite.noSites.button.accountAndSettings", value: "Account and settings", comment: "Button title. Displays the account and setting screen.")
+        static let createWPComSite = NSLocalizedString("mySite.noSites.actionSheet.createWPComSite", value: "Create WordPress.com site", comment: "Action sheet button title. Launches the flow to create a WordPress.com site.")
+        static let addSelfHostedSite = NSLocalizedString("mySite.noSites.actionSheet.addSelfHostedSite", value: "Add self-hosted site", comment: "Action sheet button title. Launches the flow to a add self-hosted site.")
+    }
+}
+
+struct NoSitesView_Previews: PreviewProvider {
+    static var previews: some View {
+        let configuration = AddNewSiteConfiguration(
+            canCreateWPComSite: true,
+            canAddSelfHostedSite: true,
+            launchSiteCreation: {},
+            launchLoginForSelfHostedSite: {}
+        )
+        NoSitesView(
+            viewModel: NoSitesViewModel(account: nil),
+            addNewSiteConfiguration: configuration
+        )
+    }
+}
```

Comment:
Does Preview function works in our project? It hasn't been functioning afaik 🤔

Reply:
@kean taught me an easy workaround for this:
- run on Xcode 15 beta 7 (not sure if it works in previous versions)
- comment out willFinishLaunching and didFinishLaunching in WordPressAppDelegate
<img width="2032" alt="Screenshot 2023-08-31 at 10 03 51" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/136d611d-02a7-4c3a-8b23-4a0f3a1fdc86">

Reply:
Interesting, that is actually a pretty useful hack